apps/prairielearn/src/middlewares/authzCourseOrInstance.js

```diff
@@ -302,7 +295,7 @@ export async function authzCourseOrInstance(req, res) {
     allow_example_course_override: false,
     ip: req.ip,
     req_date,
-    req_mode: req.cookies.pl2_requested_mode || res.locals.authz_data.mode,
+    req_mode: res.locals.authz_data.mode,
```

Comment:
Note that `req_mode` is still supported in the `select_authz_data` query in order to support the `pl_test_mode` cookie.